VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/ElementaryMethodsOfTypeInCollectionNotOverridden/ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Collections;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.General.ElementaryMethodsOfTypeInCollectionNotOverridden
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer : DiagnosticAnalyzer
+    {
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Warning;
+
+        private static readonly string Category = VSDiagnosticsResources.GeneralCategory;
+        private static readonly string Message = VSDiagnosticsResources.ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule
+            => new DiagnosticDescriptor(DiagnosticId.ElementaryMethodsOfTypeInCollectionNotOverridden, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.ObjectCreationExpression);
+
+        private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+        {
+            var objectTypeInfo = context.SemanticModel.GetTypeInfo(context.Node).Type;
+
+            var ienumerableIsImplemented = false;
+            foreach (var @interface in objectTypeInfo.AllInterfaces)
```

Comment:
This is way too brittle. Instead, use something like `Extensions.ImplementsInterface()` but you might have to remake it a little bit to no longer be an extension on `ClassDeclarationSyntax`. I don't think that's required anyway -- you can probably just work from a `INamedTypeSymbol` (or whatever `semanticModel.GetDeclaredSymbol(ClassDeclarationSyntax)` returns.